src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -34,14 +48,20 @@ public bool HasType(ResourceTypeReference typeReference)
 
             if (!typeFactories.TryGetValue(key, out var typeFactory))
             {
-                var types = TypeLoader.LoadTypes(typeReference.Namespace, typeReference.ApiVersion);
+                var types = typeLoader.LoadTypes(typeReference.Namespace, typeReference.ApiVersion);
                 typeFactory = types.Any() ? new AzResourceTypeFactory(types, typeReference.ApiVersion) : null;
                 typeFactories[key] = typeFactory;
             }
 
             return typeFactory;
         }
 
+        public ImmutableArray<ResourceTypeReference> GetAvailableTypes()
+            => availableResourceTypes.Value;
+
+        private ImmutableArray<ResourceTypeReference> GetAvailableTypesWithoutCaching()
+            => typeLoader.ListAllAvailableTypes().Keys.Select(ResourceTypeReference.Parse).ToImmutableArray();
```

Comment:
Calling `ResourceTypeReference.Parse` on 3k+ types feels like an unnecessary overhead at initialization time. Do you actually need these parsed, or will a string representation do?